packages/fbjs/src/functional/partitionObjectByKey.js

```diff
@@ -18,10 +18,10 @@ var partitionObject = require('partitionObject');
  * `whitelistObjectKeys`, but eventually keeping all the keys. Returns a tuple
  * of objects `[first, second]`.
  */
-function partitionObjectByKey(
-  source: Object,
+function partitionObjectByKey<Tv>(
+  source: {[key: string]: Tv},
   whitelist: Set<string>
-): Array<Object> {
+): [{[key: string]: Tv}, {[key: string]: Tv}] {
```

Comment:
Try `[Object, Object]`

Seems like a good compromise between the current troublesome annotation and the actual tuple type.